Add Fluent library to the website project

maxxcrawford opened the pull request:
## Summary

This PR adds Fluent (via [django-ftl](https://github.com/django-ftl/django-ftl) to the Django app

TODO: 

- [x] Header
- [x] Footer
- [x] Homepage
- [x] FAQ
- [x] Profile (logged in)
- [x] Survey Banner
- [x] ~Add Bento box strings [See Monitor bento.ftl implementation](https://github.com/mozilla/blurts-server/blob/main/locales/en/bento.ftl)~ Filed a follow-up issue here:  #822
- [x] ~Update hero images (Text is embedded)~ The strings are added to `app.ftl`, but we'll need to bust up the art to make this work fully. Filed a follow-up issue here:  #821

## Testing 

Verify English:
- Set browser language as a `en*` variant. 
- Navigate to homepage: http://localhost:8000/
- Expected: Headline should have `[EN]` string visible (See screenshot)



Verify German (non-English): 
_Note - To verify German, please checkout this commit: 2cc98dc0efac18306bb8f22e5cb6ae82f71f9f8d_
- Set browser language as a `de*` variant. 
- Navigate to homepage: http://localhost:8000/
- Expected: Headline should have `[DE]` string visible

## Screenshots

<img width="706" alt="image" src="https://user-images.githubusercontent.com/2692333/115604952-918b5d80-a2a7-11eb-86c8-eb21816e3017.png">

maxxcrawford commented on privaterelay/locales/de/privaterelay/brands.ftl:
```diff
@@ -0,0 +1,171 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+### Please read the trademarks localization guide
+### https://mozilla-l10n.github.io/styleguides/mozilla_general/index.html#brands-copyright-and-trademark
```
This directory (`privaterelay/locales/de/`) is temporary to verify Fluent works as expected. These files will be removed before merging.

maxxcrawford commented on privaterelay/locales/de/privaterelay/home.ftl:
```diff
@@ -0,0 +1,10 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
For testing purposes only.

maxxcrawford commented on privaterelay/locales/en/privaterelay/brands.ftl:
```diff
@@ -0,0 +1,171 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
```
Taken from [Bedrock](https://github.com/mozilla/bedrock/blob/master/l10n/en/brands.ftl)

Question: Should we par the list down to only strings that are used? I pared the list down considerable from Bedrock, but there are still strings that are not used.

maxxcrawford commented on privaterelay/templates/home.html:
```diff
@@ -1,15 +1,18 @@
 
 {% include "includes/messages.html" %}
 
+{% withftl bundle='privaterelay.ftl_bundles.main' %}
```
I used the [`withftl`](https://django-ftl.readthedocs.io/en/latest/usage.html#withftl) method, rather than including the file via [`ftlconf`](https://django-ftl.readthedocs.io/en/latest/usage.html#ftlconf). This can be revised as needed.

maxxcrawford commented on privaterelay/templates/includes/remaining-aliases.html:
```diff
@@ -1,17 +1,24 @@
+{% load ftl %}
+{% ftlconf bundle='privaterelay.ftl_bundles.main' %}
 
+{% remaining_free_aliases relay_addresses as remaining_free_aliases_count %}
```
This variable is set to capture the `simple_tag` function call and the output is passed as an argument to the translation string.

maxxcrawford commented on static/js/analytics.js:
```diff
@@ -153,6 +153,7 @@ function analyticsSurveyLogic() {
   switch (surveyType) {
     case "nps": {
       const notLikely = document.createElement("li");
+      // TODO: Translate these strings
```
We need to add Fluent  support to the JavaScript for this function.

groovecoder replied:
FWIW, my approach in Monitor for translating the JS-powered micro-survey responses was to add data attributes for the strings (e.g., `data-{{ msg_id }}-translated="{% ftlmsg msg_id %}"`) and then have the JS put that value into the `textContent` of the proper element.

groovecoder reviewed (changes requested):
Code looks good and works, though I haven't finished reading the django FTL docs yet. Could we move all the `ftlconf` tags up right under the `{% load ftl %}` tag? To keep things consistent and have our concerns all grouped together.

Is this ready for i18n/l10n review?

maxxcrawford commented on privaterelay/templates/faq.html:
```diff
@@ -1,61 +1,45 @@
+
 
 {% block content %}
+{% ftlconf bundle='privaterelay.ftl_bundles.main' %}
```
@groovecoder Let me confirm, but I think I needed it to be after the `{% block content %}` opening.

maxxcrawford replied:
Removed!

maxxcrawford commented on privaterelay/locales/config/pontoon.toml:
```diff
@@ -0,0 +1,15 @@
+
+basepath = ".."
+
+locales = [
+    "de",
+    "en-CA",
+    "en-GB",
+    "fr",
+]
```
No – This is where I think we set which languages we want to support. We need @peiying2 to take a look to see how we need to actually configure this before merging. 

@peiying2 Could you take a quick look to make sure we've set this up correctly? Additionally, if this needs to stay OUT of Pontoon for now, please advise if we need to remove anything.

peiying2 replied:
If this project is not ready for Pontoon, then can we exclude the file? Or since the project is not set up in Pontoon,  so nothing will be picked up there, and you can keep your work here.

peiying2 replied:
A side question, is this considered marketing  content, thus Anika and the like would be working on this once enabled in Pontoon. Or is it for the communities?

maxxcrawford replied:
> A side question, is this considered marketing content, thus Anika and the like would be working on this once enabled in Pontoon. Or is it for the communities?

I would assume it would be treated the same as [Monitor](https://pontoon.mozilla.org/projects/firefox-monitor-website/) or [Facebook Container](https://pontoon.mozilla.org/projects/facebook-container/)

groovecoder replied:
@maxxcrawford - I would say that we should pull this list from an env var like we do for Monitor, but that just shifts the responsibility from an app code change to an ops code change, so I guess 🤷‍♂️ . Maybe best to keep it in our own code.

maxxcrawford replied:
@peiying2 Excellent. Thanks for the insight on this.  

@groovecoder I recommend we remove the file all together and let it supported locales be set in Pontoon.

maxxcrawford replied:
This file has been removed.

maxxcrawford commented on privaterelay/views.py:
```diff
@@ -55,7 +56,9 @@
-    return render(request, 'home.html')
+    return render(request, 'home.html', {
+        'ftl_bundle': main_bundle,
+    })
```
This is some copy-pasta left over from the docs when I first installed it. I think this can be removed (as the other pages don't need them to work). I'll confirm and revise as necessary.

maxxcrawford replied:
Removed!

groovecoder commented on privaterelay/views.py:
```diff
@@ -37,6 +37,7 @@
 )
 from emails.utils import get_post_data_from_request
 
+from .ftl_bundles import main as main_bundle
```
Don't need this line anymore.

maxxcrawford replied:
Fixed!

groovecoder commented on privaterelay/locales/l10n-pontoon.toml:
```diff
@@ -0,0 +1,7 @@
+basepath = "."
+
+[[includes]]
+    path = "configs/pontoon.toml"
```
This path `configs/pontoon.toml` doesn't match the directory structure `config/pontoon.toml`?

maxxcrawford replied:
Fixed!

peiying2 commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+fx-pocket = { -brand-name-pocket }
+mobileCloseBentoButtonTitle = Close menu
+
+## Home Page - https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/
```
This is not needed. The product staging info will provided in Pontoon on project dashboard.

maxxcrawford replied:
Cool. I pulled this convention from the Bedrock project. Should I remove all of the URLs?

maxxcrawford replied:
Removed!

peiying2 commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account…
+hero-image-copy-protect-html = …so you can <strong>protect your real email</strong> from tracking and spam.
+hero-image-copy-control-html = Now <em>you're in control</em> of what's arriving to your inbox!
+
```
Remove extra line space (nit pick)

maxxcrawford replied:
Removed!  I removed some additional extra lines throughout the file too.

peiying2 reviewed (changes requested):
Structure of  the files:
- This app.ftl source file is for `en` or `en-US`. Make sure to use the correct locale code. The  toml file refers to `en`. The difference between en and en-US is that  en-US is treated like a locale, and it allows you to add US specific content while en is the source for all other languages.
- File path/directory structure: Recommend to use: {productname}/locales/{locale}/app.ftl. Right now, locaels/{locale}/**privatereplay**/app.ftl. If you have more  than one file than app.ftl, you can list them all under locales/{locael}/...
- Are we going to have an l10n repo or is localized content going to be part of this product code.
- If there is no interest in using vendor for the localization, we can do away toml files which configures which locales goes to where (pontoon or vendor platform). If this is for community to  localize, we can remove the pontoon.toml file.

Here is the summary of the common issues identified in the app.ftl file itself:
- Brands section, 1). add parameter for Firefox Account entry depending on the function of the term. 2). Consider adding Relay Premium and Premium as terms and reference them in the file
- Use curly quote throughout: single and double
- Use `extension` instead of `add-on`.

flodolo commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+## Meta Data
+
+meta-title = { -brand-name-firefox-relay }
```
Note: this string is OK, there was some misunderstanding between Peiying and I.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+hero-image-copy-trust = Can you even trust this company with your personal email?
+hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account…
+hero-image-copy-protect-html = …so you can <strong>protect your real email</strong> from tracking and spam.
+hero-image-copy-control-html = Now <em>you're in control</em> of what's arriving to your inbox!
```
It's [a convention used](https://django-ftl.readthedocs.io/en/latest/usage.html#html-escaping) from our Fluent python lib to escape HTML into strings. If there is a better way to handle this, I'm happy to revisit.

flodolo replied:
I think you're talking about two different things? This is about consistently using typographic apostrophes instead of straight quotes, i.e. not `what's`, but `what’s`. Same applies in the rest of the file, and for `""` vs `“”` (**not** when used in markup).

maxxcrawford replied:
@flodolo Understood about the single/double quotes. 

I was answering this question with my reply: 
> why these last three string IDs contain -html. Just curious.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+# Variables:
+#   $url (url) - https://addons.mozilla.org/firefox/addon/private-relay/
+#   $attrs (string) - specific attributes added to external links
+faq-question-1-answer-html = While { -brand-name-relay } does not filter for spam, our email partner Amazon SES does block spam and malware. If { -brand-name-relay } forwards messages you don’t want, you can update your { -brand-name-relay } settings to block messages from the alias forwarding them. <br/><br/> If you see a broader problem of unwanted email from all of your aliases, please <a  hre ...
```
Broken up. I also broke up other lines as well.

maxxcrawford replied:
(and removed the double space)

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-headline-manage-domain = Manage your Domain aliases
+profile-supports-email-forwarding = { -brand-name-firefox-relay } supports email forwarding (including attachments) of email up to 150KB in size
+profile-promo-premium-headline = Go premium for more features.
+profile-promo-premium-copy = With { -brand-name-relay } Premium, you get your own email domain.
```
I don't think it is – or at least, it isn't yet. Let me confirm. 

@groovecoder Should we use "Firefox Relay Premium" here?

groovecoder replied:
No - neither "Firefox Relay Premium" nor "Relay Premium" will be a product name. "premium" is just a description that the particular feature requires "premium" access, which will be paid.

flodolo replied:
Let me rephrase the question: is the expectation that Premium will stay in English, or it can be translated without limitations?

maxxcrawford replied:
Translated without limitations. It's to let them know it's extra features.

maxxcrawford replied:
Quick update – I reached out to our copywriter to address the entire "Relay Premium" issue. It'll avoid any sort of branded name so the string can be fully translated (besides "Relay"). I'll updated strings tomorrow.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-label-saved = Label saved!
+profile-label-generate-new-alias = Generate New Alias
+profile-label-delete = Delete
+profile-label-forward-emails = Forward emails to
```
Whatever the users' email address is will populate here. There was not a colon originally, so I kept it the same. Should we add on, or should I add a comment for context?

<img width="427" alt="image" src="https://user-images.githubusercontent.com/2692333/118047226-16661600-b340-11eb-83f3-95a70105ce84.png">

peiying2 replied:
Add a comment will be good, `This is followed by an email address`

maxxcrawford replied:
Also, I'll add the `:`.  We should aim for consistency, and `Created:` has it.

maxxcrawford replied:
Done.

peiying2 commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+how-it-works-step-2-headline = Create a new alias
+how-it-works-step-2-copy = As you browse, the { -brand-name-relay } icon will appear in form fields where sites ask for your email address. Select it to generate a new, random address that ends in @relay.firefox.com. { -brand-name-relay } will forward messages to the primary email address associated with your account.
+how-it-works-step-3-headline = Manage your account
+how-it-works-step-3-copy = Sign in to the { -brand-name-relay } website to keep track of the aliases you’ve created.  If you find that one receives spam or unwanted messages, you can block all messages or even delete the alias, right from the management page.
```
Single space two sentences `...you’ve created. If you find...`

maxxcrawford replied:
I broke this line up into separate lines, and removed the extra space. Please let me know if my new line breaks are causing any issues.

flodolo commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+home-hero-headline = Hide your real email address to help protect your identity
+home-hero-copy = { -brand-name-firefox-relay } makes it easy to create aliases, randomly generated email addresses that forward to your real inbox. Use it to protect your online accounts - and your identity - from hackers. Sign in with your { -brand-name-firefox-account } to get started.
+home-hero-cta = { nav-profile-sign-in }
```
Note: this comment (and others) is marked as resolved but it really isn't? My guess is that you didn't push your changes?

maxxcrawford replied:
@flodolo Yup. Just pushed it up. I'll refrain from marking resolved until the commit is up officially.

flodolo commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+survey-option-somewhat-disappointed = Somewhat disappointed
+survey-option-very-disappointed = Very disappointed
+survey-option-very-likely = Very likely
+survey-option-not-likely = Not likely
```
Please keep an empty line at the end of the file

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+hero-image-copy-trust = Can you even trust this company with your personal email?
+hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account…
+hero-image-copy-protect-html = …so you can <strong>protect your real email</strong> from tracking and spam.
```
These strings are part of the image gallery on the homepage (currently, this text is embedded in the images - #821). It will be resolved in a follow-up PR.

peiying2 replied:
If used in an image, make sure there is space between the punctuation and the first word after it so it looks like `… so you can`

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+## Evergreen Survey (displayed on the profile page) - https://dev.fxprivaterelay.nonprod.cloudops.mozgcp.net/accounts/profile/
+
+survey-question-1 = On a scale from 1-10, how likely are you to recommend Relay to a friend or colleague?
```
Fixed on all the survey entries!

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-label-delete = Delete
+profile-label-forward-emails = Forward emails to
+profile-label-first-emailed = First emailed:
+profile-label-created = Created:
```
Context added.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-label-created = Created:
+profile-label-details-show = Show Details
+profile-label-details-hide = Hide Details
+profile-label-forwarding = forwarding
```
It's a label for a switch to turn off/on. It's lowercase per the design. 

<img width="242" alt="image" src="https://user-images.githubusercontent.com/2692333/118147541-4b707800-b3d5-11eb-91d9-138d431fdc0b.png">

maxxcrawford replied:
Added the comment:  `# This string is a label for a toggle (on/off) switch`

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-label-generate-new-alias = Generate New Alias
+profile-label-delete = Delete
+profile-label-forward-emails = Forward emails to
+profile-label-first-emailed = First emailed:
```
Context added.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+profile-label-forwarded = Forwarded
+profile-label-cancel = Cancel
+profile-blocked-copy = { -brand-name-firefox-relay } will delete messages before they can reach your inbox when you select blocking for this alias.
+profile-forwarded-copy-html = { -brand-name-firefox-relay } will send messages to your inbox when you select forwarding for this alias. <br /><span class="fwd-note"><span class="bold ff-Met">Note:</span> Email (including attachments) larger than 150KB are not currently supported and will not be forwarded.
```
I can break up this HTML into three separate strings: 
```
profile-forwarded-copy: { -brand-name-firefox-relay } will send messages…
profile-forwarded-note: Note: 
profile-forwarded-note-copy: Email (including attachments) larger…

maxxcrawford replied:
Broken up (and FWIW re-wrote the logic to show/hide this to not be JS dependent)

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+banner-premium-cta = Buy Premium
+banner-choose-subdomain-headline = Choose your own domain
+banner-choose-subdomain-copy = You can choose your own domain for "wildcard" aliases. NOTE: YOU CANNOT CHANGE THIS LATER!
+banner-choose-subdomain-label = Your domain is:
```
Yes. Context added.

pdehaan commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,226 @@
+profile-label-cancel = Cancel
+profile-blocked-copy = { -brand-name-firefox-relay } will delete messages before they can reach your inbox when you select blocking for this alias.
+profile-forwarded-copy-html = { -brand-name-firefox-relay } will send messages to your inbox when you select forwarding for this alias. <br />
+    <span class="fwd-note"><span class="bold ff-Met">Note:</span> Email (including attachments) larger than 150KB are not currently supported and will not be forwarded.
```
Can you please confirm these HTML tags?
I see two opening `<span` tags but only one closing `</span>` tag.
My guts are saying this should be a combined `<span class="fwd-note bold ff-Met">` tag.

peiying2 replied:
Can we do away the CSS code in the string? If not, yeah, if we can consolidate them, that's better/cleaner. so it won't risk being broken up.

maxxcrawford replied:
I updated this entire string structure. It's now a standard `<strong>` tag, and no HTML is included in the string.

pdehaan commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,226 @@
+banner-download-firefox-headline = { -brand-name-relay } is even better in { -brand-name-firefox }
+banner-download-firefox-copy = The { -brand-name-relay } extension for { -brand-name-firefox-browser } makes creating aliases even easier.
+banner-download-firefox-cta = Get { -brand-name-firefox }
+banner-download-install-add-on-headline = Looks like you haven’t installed the add-on…yet!
```
Q: Should there be a space between the "&hellip;" and "yet!"?

maxxcrawford replied:
Context for the string. I think we should remove it too. 

![image](https://user-images.githubusercontent.com/2692333/118539665-500a9880-b715-11eb-8b06-0a0afa9d89de.png)

maxxcrawford replied:
Revised: 
![image](https://user-images.githubusercontent.com/2692333/118539793-78929280-b715-11eb-836e-d9aa14ef3a75.png)

pdehaan commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+banner-download-firefox-cta = Get { -brand-name-firefox }
+banner-download-install-add-on-headline = Looks like you haven't installed the add-on... yet!
+banner-download-install-add-on-copy = The { -brand-name-relay } extension for { -brand-name-firefox-browser } makes creating aliases even easier.
+banner-download-install-add-on-cta = Get the add-on for { -brand-name-firefox }
```
Excellent question! I see "extension" used in strings 4 times, and "add-on" twice (here and L170 above).  Probably best to standardize on a single term.

maxxcrawford replied:
I know that the official term is "Add-on" based on AMO, Bedrock — but over in about:addons, it says "Extensions".

flodolo replied:
"Add-on" is an umbrella term that includes themes, extensions, and plugins. It's also being deprecated in Firefox in favor of "extensions", with a few exception.

maxxcrawford replied:
Extension it is! :)

maxxcrawford replied:
Updated.

pdehaan commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,226 @@
+fx-mobile = { -brand-name-firefox-browser } for Mobile
+fx-monitor = { -brand-name-firefox-monitor }
+fx-pocket = { -brand-name-pocket }
+mobileCloseBentoButtonTitle = Close menu
```
This looks non-kebab-case, whereas everything else looks kebaby. 🤷

peiying2 replied:
I missed this one.

pdehaan commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,226 @@
+faq-question-6-question = What happens if { -brand-name-mozilla } shuts down the { -brand-name-firefox-relay } service?
+faq-question-6-answer = We will give you advance notice that you need to change the email address of any accounts that are using { -brand-name-relay } aliases.
+faq-question-7-question = What if an email sent to my alias contains an attachment?
+faq-question-7-answer = We now support attachment forwarding. However, there is a 150KB limit for email forwarding using { -brand-name-relay }. 
```
Possibly a Q for @flodolo and @peiying2, but should the `150KB` be a variable? I think I found 4&times; instances in this .ftl file (here, L128, and L154), meaning that if we change that limit to 300KB at some point, we'd invalidate 4 strings per locale. Perhaps possibly using a `{ -email-size-limit }` variable and change it in one place per locale.

flodolo replied:
Definitely not a bad idea.

peiying2 replied:
Agreed. Don't want to update the string when the number changes.

maxxcrawford replied:
Awesome. I'll add it.

maxxcrawford replied:
Added a comment for additional context too. 

```
# The file size limit for emails forwarded through the Relay service. KB is an abbreviation for kilobyte.  
-email-size-limit = 150KB
```

peiying2 commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -55,8 +56,9 @@ mobileCloseBentoButtonTitle = Close menu
-home-hero-copy = { -brand-name-firefox-relay } makes it easy to create aliases, randomly generated email addresses that forward to your real inbox. Use it to protect your online accounts - and your identity - from hackers. Sign in with your { -brand-name-firefox-account } to get started.
-home-hero-cta = { nav-profile-sign-in }
+home-hero-copy = { -brand-name-firefox-relay } makes it easy to create aliases, randomly generated email addresses that forward to your real inbox.
+    Use it to protect your online accounts - and your identity - from hackers. Sign in with your { -brand-name-firefox-account } to get started.
```
This needs to have a string ID, maybe make the string IDs as 
- `home-hero-copy-1 =`
- `home-hero-copy-2 =`

Can you split it up like this? I think it is ok to combine this one. The ones with line breaks can be broken up.

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+-brand-name-firefox = Firefox
+-brand-name-firefox-relay = Firefox Relay
+-brand-name-relay = Relay
+-brand-name-firefox-account = Firefox Account
```
@peiying2 What does `{ -fxaccount-brand-name(nbroken: "frase") }` mean in your example?

flodolo replied:
https://mozilla-l10n.github.io/localizer-documentation/tools/fluent/basic_syntax.html#terms-and-parameterized-terms

maxxcrawford replied:
Revised! I made the lowercase version the default. There were only 3 places (all within the profile component) with it was supposed to be capitalized.

peiying2 commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,233 @@
+    If you find that one receives spam or unwanted messages, you can block all messages or even delete the alias, 
+    right from the management page.
+hero-image-copy-trust = Can you even trust this company with your personal email?
+hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account…
```
Should have space between the last word and the punctuation so it looks like `every new account …`

maxxcrawford replied:
@peiying2 Can you confirm this is the change you want to see? 
```suggestion
hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account …
```

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+# Variables:
+# $number (number) - User-set subdomain
+remaining-aliases-promo-label = { $number } aliases. You can make unlimited!
```
Revised (and confirmed locally). Thanks!

maxxcrawford commented on privaterelay/locales/en-us/privaterelay/app.ftl:
```diff
@@ -0,0 +1,208 @@
+
+# Variables:
+# $number (number) - Number of remaining aliases. This will always be more than 1. 
+remaining-aliases-label-plural-html = <span>{ $number }</span> remaining aliases
```
Revised/removed!

maxxcrawford commented:
@peiying2 

- This app.ftl source file is for en or en-US. Make sure to use the correct locale code. The toml file refers to en. The difference between en and en-US is that en-US is treated like a locale, and it allows you to add US specific content while en is the source for all other languages. **I removed the TOML file, and will leave it as en-US. (The website breaks on other languages when I have only a /en/ file.**
- File path/directory structure: Recommend to use: {productname}/locales/{locale}/app.ftl. Right now, locaels/{locale}/privatereplay/app.ftl. If you have more than one file than app.ftl, you can list them all under locales/{locael}/... **I removed the extra {productname} directory. It's now `/privaterelay/locales/en-us/app.ftl`**
- Are we going to have an l10n repo or is localized content going to be part of this product code. **I do not know the answer to this. Do you have a recommendation? @groovecoder Do you have an opinion? @moz-astults?**
- If there is no interest in using vendor for the localization, we can do away toml files which configures which locales goes to where (pontoon or vendor platform). This is a product so it will go to Pontoon, to be localized by the communities. **Copy that. Files removed!**

peiying2 commented:
@maxxcrawford Thanks for the update. I realize the comments in this PR is getting very long, so hopefully your questions are answered and additional comments/suggestion for change is updated too. 

One more thing regarding the changes you made after our suggestions to breaking up sentences that contained <br> markups. After the breakup, each string needs to have a string ID. If you want them to group together, you can add an extra element like same-paragraph-copy-a, same-paragraph-copy-b, same-paragraph-copy-c, for example, so you can tell these strings are grouped together.

maxxcrawford commented:
> @maxxcrawford Thanks for the update. I realize the comments in this PR is getting very long, so hopefully your questions are answered and additional comments/suggestion for change is updated too.
> 
> One more thing regarding the changes you made after our suggestions to breaking up sentences that contained
> markups. After the breakup, each string needs to have a string ID. If you want them to group together, you can add an extra element like same-paragraph-copy-a, same-paragraph-copy-b, same-paragraph-copy-c, for example, so you can tell these strings are grouped together.

I revised the string you pointed to in this commit: https://github.com/mozilla/fx-private-relay/pull/815/commits/87cfcd9c18ab1534889e95febc706b0cd765c397

maxxcrawford commented:
> ID `b` contains `html` but ID `a` doesn't. Let's standardize the naming convention.

The second string contains HTML – but the first doesn't. I'll update them both to use `-html`.

peiying2 commented on privaterelay/locales/en-us/app.ftl:
```diff
@@ -0,0 +1,239 @@
+    right from the management page.
+hero-image-copy-trust = Can you even trust this company with your personal email?
+hero-image-copy-unique-html = <strong>Use a unique relay address</strong> for every new account …
+hero-image-copy-protect-html = …so you can <strong>protect your real email</strong> from tracking and spam.
```
should there a space after the ellipsis

maxxcrawford replied:
I added a space. Happy to match the usage you defined: 

> Ellipsis usage: space before it when it ends a sentence, and a space after it when starting a sentence as in row90.

peiying2 commented:
1). `en-us` should be `en-US`: lower case is locale code, all cap is country code. 
2). Ellipsis usage: space before it when it ends a sentence, and a space after it when starting a sentence as in [row90](https://github.com/mozilla/fx-private-relay/pull/815/files#diff-e14e064050f9c7c508129a37ef6c96054af00b08bb421ba06ca0adf2378f16d5R90).
3). I assume the decision the repo for l10n is yet to be decided.

flodolo commented on privaterelay/locales/en-us/app.ftl:
```diff
@@ -0,0 +1,239 @@
+-brand-name-pocket = Pocket
+-brand-name-firefox-account = 
+    { $capitalization ->
+        *[lowercase] Firefox account
```
nit: align the parenthesis, not the * sign

```
-brand-name-firefox-account = 
    { $capitalization ->
       *[lowercase] Firefox account
        [uppercase] Firefox Account
    }
```

flodolo commented on privaterelay/locales/en-us/app.ftl:
```diff
@@ -0,0 +1,239 @@
+    }
+    
+# The file size limit for emails forwarded through the Relay service. KB is an abbreviation for kilobyte.  
+-email-size-limit = 150KB
```
This doesn't really scale.

One issue is that terms should be used for things like brand/product names, where you might need private attributes. For this you can use a normal message (standard messages can be referenced in other messages, but also used stand-alone in code).

Second issue: what happens if you change the value? You need to update this string, which requires a new ID, and breaks all existing translations (which will require new IDs).

This should be

```
# Variables:
#   $size (number): maximum size for attachments
email-size-limit = { $size } KB
```

Even better, so we don't break if we switch to some MB.

```
# String used to display the attachment limit, e.g. 150 KB
# Variables:
#  $size (number): maximum size for attachments
#  $unit (string): unit of measurement (e.g. KB)
email-size-limit = { $size } { $unit }
unit-kb = KB
```

Questions on the English:
* Shouldn't there be a space between number and unit?
* `kB` or `KB`?

This should also move closer to the string that uses it.

maxxcrawford replied:
> Shouldn't there be a space between number and unit? 
Yes. Adding. 

> `kB` or `KB`?

KB because this file size limit is in kilobytes, rather than bits. (@groovecoder – Can you confirm?) 

---

I want to use the two-argument variable string. How do I reference `unit-kb`? In the example below I set the unit value in the `HTML`. 

**In the FTL:** 
```
faq-question-7-answer = We now support attachment forwarding. However, there is a { -email-size-limit } limit for email forwarding using { -brand-name-relay }. Any emails larger than { -email-size-limit } will not be forwarded.
```

**In the HTML:** 
```
<p class="faq-answer">{% ftlmsg 'faq-question-7-answer' %}</p>
```

```
<p class="faq-answer">{% ftlmsg 'faq-question-7-answer' size='150' unit='KB' %}</p>
```

maxxcrawford replied:
I passed arguments everywhere the string is referenced. I have the `unit-kb` variable defined in the FTL file, but not used anywhere. Please advise.

flodolo replied:
> I want to use the two-argument variable string. How do I reference `unit-kb`? In the example below I set the unit value in the `HTML`.

I'm afraid I'm out of my depth when it comes to the actual code, and I'm not sure who else is familiar with this library at this point.

Worst case scenario, I would `KB` to `unit`, because having the message reference (i.e. `email-size-limit = { $size } { mb-unit }`) would have the same scalability/maintainability problem.

flodolo replied:
Not sure if [this](https://django-ftl.readthedocs.io/en/latest/usage.html#lazy-translations) helps (can you store the unit-kb translation in variable, and pass that as the argument? I have no clue…)

maxxcrawford replied:
Okay – Talking with the team, this is okay as is. We are not planning on changing the limit any time in the near/far future. It is not on our roadmap.

flodolo replied:
Not the first time we've been bitten by this approach…

Out of curiosity, the suggestion I linked to in the previous comment doesn't work? I'd still prefer to have that fully localizable and future-proof, if possible. If not, you should remove the "KB" string, since nothing is using it.

maxxcrawford replied:
Oh! Yeah – the previous comment didn't work because this is visible without any user interaction. The lazy function is for modals and things that only show when the user does an action to display them. However, this could be used for other strings in this application (Namely – all the strings used in `/includes/modal.html` and some in the `profile.html`). 

I'll file an issue to revist this. 

(also I'm removing that unused variable for now)

flodolo commented on privaterelay/locales/en-us/app.ftl:
```diff
@@ -0,0 +1,239 @@
+# Variables:
+#   $url (url) - https://addons.mozilla.org/firefox/addon/private-relay/
+#   $attrs (string) - specific attributes added to external links
+faq-question-2-answer-html = Some sites may not accept an email address that includes a subdomain (ie, the “relay” portion of @relay.firefox.com) and others have stopped accepting all addresses except those from Gmail, Hotmail, or Yahoo accounts. 
```
ie or i.e.?

maxxcrawford replied:
"i.e." is the correct usage from what I can gather.

maxxcrawford replied:
(Updated to i.e.)

maxxcrawford commented:
> 1). `en-us` should be `en-US`: lower case is locale code, all cap is country code.
> 2). Ellipsis usage: space before it when it ends a sentence, and a space after it when starting a sentence as in [row90](https://github.com/mozilla/fx-private-relay/pull/815/files#diff-e14e064050f9c7c508129a37ef6c96054af00b08bb421ba06ca0adf2378f16d5R90).
> 3). I assume the decision the repo for l10n is yet to be decided.

1. Updated the folder to the `en-US` correct capitalization 
2. Updated ellipsis spacing. Quick reminder that this is part of an outstanding issue (#821) with the strings being embedded in the images currently. 
3. No decision yet. As this isn't immediately being localized, I think we can defer this decision immediately.

maxxcrawford commented:
@flodolo @peiying2 I think this is good to merge in, unless y'all wanna review anything further!

peiying2 commented:
> @flodolo @peiying2 I think this is good to merge in, unless y'all wanna review anything further!

I don't  have any additional comment. I want to make sure the locale name is fixed. It should be: en-US.

maxxcrawford commented:
> > @flodolo @peiying2 I think this is good to merge in, unless y'all wanna review anything further!
> 
> I don't have any additional comment. I want to make sure the locale name is fixed. It should be: en-US.

Great catch, @peiying2. I pushed an update to revise it. 

https://github.com/mozilla/fx-private-relay/blob/813-add-fluent/privaterelay/locales/en-US/app.ftl

maxxcrawford commented:
@groovecoder I think we're ready to merge – could you please re-review one more time? (I know we're already approved by you, but wanted to double confirm)

groovecoder commented on privaterelay/views.py:
```diff
@@ -37,7 +37,6 @@
 )
 from emails.utils import get_post_data_from_request
 
-
```
F this line in particular.

groovecoder commented:
Code and spot-check looks good. I found an un-related bug I'll file separately.
